### PR TITLE
Added type definitions for noloopfor

### PR DESCRIPTION
**Pull request by david-fernando**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/noloopfor/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for noloopfor 2.0
+// Project: https://github.com/david-fernando/noLoopFor
+// Definitions by: David Fernando <https://github.com/david-fernando/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+export function loop(value: number, target: number, callback: any): any;
+export function loopStep(value: number, target: number, step: number, callback: any): any;
+export function decrementLoop(value: number, target: number, callback: any): any;
+export function decrementStep(value: number, target: number, step: number, callback: any): any;
```
Comment (DanielRosenwasser) [inline types/noloopfor/index.d.ts:8]: Seems like these should all take actual callback types

```suggestion
export function loop(value: number, target: number, callback: (index: number) => void): any;
export function loopStep(value: number, target: number, step: number, callback: (index: number) => void): any;
export function decrementLoop(value: number, target: number, callback: (index: number) => void): any;
export function decrementStep(value: number, target: number, step: number, callback: (index: number) => void): any;
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

File: types/noloopfor/tslint.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+    "whitespace": false
```
[the remark this answers is not in the document]
Comment (david-fernando) [reply types/noloopfor/tslint.json]: This rule exists to that the test does not require white space in the code.

File: types/noloopfor/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// Type definitions for noloopfor 2.0
+// Project: https://github.com/david-fernando/noLoopFor
+// Definitions by: David Fernando <https://github.com/david-fernando/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+export function loop(value: number, target: number, callback: (index: number) => void): any;
+export function loopStep(value: number, target: number, step: number, callback: (index: number) => void): any;
+export function decrementLoop(value: number, target: number, callback: (index: number) => void): any;
+export function decrementStep(value: number, target: number, step: number, callback: (index: number) => void): any;
```
Comment (peterblazejewicz) [inline types/noloopfor/index.d.ts:8]:
```suggestion
export function loop(value: number, target: number, callback: (index: number) => void): void;
export function loopStep(value: number, target: number, step: number, callback: (index: number) => void): void;
export function decrementLoop(value: number, target: number, callback: (index: number) => void): void;
export function decrementStep(value: number, target: number, step: number, callback: (index: number) => void): void;
```
were those `any` by design ther? Those are just undefined, so `void`

File: types/noloopfor/tslint.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "dtslint/dt.json",
+  "rules": {
+      "whitespace": false
+  }
```
Comment (peterblazejewicz) [inline types/noloopfor/tslint.json:5]:
```suggestion
  "extends": "dtslint/dt.json"
```
why do you need this exclusion?
https://github.com/DefinitelyTyped/DefinitelyTyped#linter-tslintjson

**Review by peterblazejewicz — APPROVED** [review]
LGMT!
@david-fernando thx!

**Pull request merged**